src/main/java/marquez/db/dao/JobDAO.java

```diff
@@ -1,14 +1,64 @@
 package marquez.db.dao;
 
+import java.sql.Timestamp;
 import java.util.List;
+import java.util.Optional;
 import marquez.api.Job;
+import marquez.api.Ownership;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.CreateSqlObject;
+import org.jdbi.v3.sqlobject.SqlObject;
 import org.jdbi.v3.sqlobject.customizer.Bind;
 import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public interface JobDAO {
-  @SqlQuery("SELECT * FROM jobs WHERE id = :id")
-  Job findById(@Bind("id") long id);
+public interface JobDAO extends SqlObject {
+  static final Logger LOG = LoggerFactory.getLogger(JobDAO.class);
+
+  @CreateSqlObject
+  OwnershipDAO createOwnershipDAO();
+
+  default void insert(final Job job) {
+    try (final Handle handle = getHandle()) {
+      handle.useTransaction(
+          h -> {
+            final int jobId =
+                h.createUpdate(
+                        "INSERT INTO jobs (name, nominal_time, category, description)"
+                            + " VALUES (:name, :nominalTime, :category, :description)")
```

Comment:
Will this have update behavior instead of insert if a row with the same primary key is already present?

Reply:
For convenience, I added `JobDAO.updateOwnership()`, but it probably doesn't make sense for a job to update it's ownership. We can simply have the `OwnershipDAO` update `job.current_ownership.` and handle all the logic around changing ownership over time.